### PR TITLE
Feature/update arith opcodes

### PR DESCRIPTION
**Pull request by invocamanman**
- Udpate addMod/MulMod
- SignExtend should be finished, it uses Js for calculating the exponentiation

### CODE REVIEW

File: main/opcodes.zkasm [hunk 1]
```diff
@@ -216,50 +209,145 @@ opADDMOD:
...
+    $ => A          :LT
+    1               :ASSERT
     C               :MSTORE(SP++)
-    1024 - SP       :JMPN(stackOverflow)
-    GAS-8 => GAS    :JMPN(outOfGas)
                     :JMP(readCode)
 
 opMULMOD:
-
     %MAX_CNT_ARITH - CNT_ARITH - 2 :JMPN(outOfCounters)
```
Comment (ignasirv) [inline main/opcodes.zkasm:249]: After all this new logic we may have to allocate more counters for this opcode

File: main/opcodes.zkasm [hunk 2]
```diff
@@ -216,50 +209,145 @@ opADDMOD:
     %MAX_CNT_BINARY - CNT_BINARY - 3 :JMPN(outOfCounters)
```
Comment (ignasirv) [inline main/opcodes.zkasm:209]: After all this new logic we may have to allocate more counters for this opcode

**Review by krlosMata — APPROVED** [review]
:smile_cat:

**Review by laisolizq — APPROVED** [review]
:smile_cat:

**Review by ignasirv — APPROVED** [review]

**Pull request merged**